cli/tfsource/types.go

```diff
@@ -43,9 +45,23 @@ func (src *TerraformSource) String() string {
 // so the same file path (/foo/bar) is always considered the same version. See also the encodeSourceName and
 // ProcessTerraformSource methods.
 func (terraformSource TerraformSource) EncodeSourceVersion() string {
+	if IsLocalSource(terraformSource.CanonicalSourceURL) {
+		hash, err := terraformSource.HashLocalDir()
+		fmt.Printf("\nLocal info!\n\tLocal hash: %s\n\tCanonical URL: %s\n\tCanonical path: %s\n\tWorking Dir: %s\n", hash, terraformSource.CanonicalSourceURL, terraformSource.CanonicalSourceURL.Path, terraformSource.WorkingDir)
+		if err == nil {
+			return hash
+		}
+	}
 	return util.EncodeBase64Sha1(terraformSource.CanonicalSourceURL.Query().Encode())
 }
 
+func (terraformSource TerraformSource) HashLocalDir() (string, error) {
+	if !IsLocalSource(terraformSource.CanonicalSourceURL) {
+		return "", fmt.Errorf("is not a local directory")
+	}
+	return dirhash.HashDir(terraformSource.CanonicalSourceURL.Path, "hi", dirhash.DefaultHash)
```

Comment:
The second parameter here is a prefix. Not sure what to use here.

Ref: [HashDir](https://pkg.go.dev/golang.org/x/mod/sumdb/dirhash#HashDir)

Reply:
Changed the prefix to `prefix`